### DISCUSSION TITLE
How smart is Prisma's use of connection pools?

### DISCUSSION THREAD

**Discussion by jimsimon** [prisma/prisma#3966] [Q&A] [answered]
I'm looking into using Prisma for a multi-tenant application using a hybrid of the tenant per database and multiple tenants per database approach. I plan on using Postgres schemas for the multiple tenants per database part. I have two questions regarding how connection pooling would work:

1. If I create multiple clients with the exact same connection string, will Prisma use the same connection pool for them?
2. If I create multiple clients with identical connection strings except for the schema (i.e. `postgres://db:5432/myapp?schema=tenant1` and `postgres://db:5432/myapp?schema=tenant2`), will this result in a connection pool per tenant or is Prisma smart enough to realize this is the same database server?

My assumption is that 1 will end up with a single connection pool and 2 will end up with multiple connection pools.

**Comment by jimsimon** [comment, marked as answer]
I was ultimately able to test this myself. Each instance of PrismaClient has its own connection pool regardless of the connection string. So I ended up with two pools even when using the same string with two instances.

**Reply by nikolasburk** [reply]
Thanks for validating your own hypotheses 😄 

Also, if you're interested in building multi-tenant applications with Prisma I'd definitely recommend to check out this library by @Errorname: https://github.com/Errorname/prisma-multi-tenant 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
